errbot/backends/slack.py

```diff
@@ -135,14 +147,19 @@ def channelname(self):
         if self._channelid is None:
             return None
 
-        channel = self._sc.server.channels.find(self._channelid)
+        if self._channelname:
+            return self._channelname
+
+        channel = [channel for channel in self._webclient.channels_list() if channel['id'] == self._channelid][0]
         if channel is None:
             raise RoomDoesNotExistError(f'No channel with ID {self._channelid} exists.')
-        return channel.name
+        if not self._channelname:
+            self._channelname = channel['name']
+        return self._channelname
 
     @property
     def domain(self):
-        return self._sc.server.domain
+        raise NotImplemented()
```

Comment:
Can this be read from [scopes/identity.team:read:user](https://api.slack.com/scopes/identity.team:read:user)?